core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -91,27 +92,27 @@ public class TestTableMetadata {
   @Test
   public void testJsonConversion() throws Exception {
     long previousSnapshotId = System.currentTimeMillis() - new Random(1234).nextInt(3600);
+
+    String manifestList =
+        createManifestListWithManifestFile(previousSnapshotId, null, "file:/tmp/manifest1.avro");
     Snapshot previousSnapshot =
         new BaseSnapshot(
-            previousSnapshotId,
-            null,
-            previousSnapshotId,
-            null,
-            null,
-            null,
-            ImmutableList.of(
-                new GenericManifestFile(localInput("file:/tmp/manfiest.1.avro"), SPEC_5.specId())));
+            0, previousSnapshotId, null, previousSnapshotId, null, null, null, manifestList);
+
     long currentSnapshotId = System.currentTimeMillis();
+    manifestList =
+        createManifestListWithManifestFile(
+            currentSnapshotId, previousSnapshotId, "file:/tmp/manifest2.avro");
     Snapshot currentSnapshot =
         new BaseSnapshot(
+            0,
             currentSnapshotId,
             previousSnapshotId,
             currentSnapshotId,
             null,
             null,
             7,
-            ImmutableList.of(
-                new GenericManifestFile(localInput("file:/tmp/manfiest.2.avro"), SPEC_5.specId())));
+            manifestList);
```

Comment:
the previous test was referring to a single manifest file which I'm doing as well via `createManifestListWithManifestFile()`. It doesn't seem to matter in these tests particularly whether the manifest list at this point contains only `/tmp/manifest2.avro` or `/tmp/manifest1.avro` + `/tmp/manifest2.avro`, so just wanted to mention this